metadataStore/test/test_commands.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+import uuid
+
+import mongoengine
+import mongoengine.connection
+from mongoengine.context_managers import switch_db
+
+from nose.tools import make_decorator
+from nose.tools import assert_equal
+
+
+from metadataStore.database import (BeamlineConfig, EventDescriptor,
+                                    Event, Header)
+import metadataStore.commands as mdsc
+
+db_name = str(uuid.uuid4())
+dummy_db_name = str(uuid.uuid4())
+
+
+def setup():
+    # need to make 'default' connection to point to no-where, just to be safe
+    mongoengine.connect(dummy_db_name)
+    # connect to the db we are actually going to use
+    mongoengine.connect(db_name, alias='test_db')
+
+
+def teardown():
+    conn = mongoengine.connection.get_connection('test_db')
+    conn.drop_database(db_name)
+    conn.drop_database(dummy_db_name)
+
+
+def context_decorator(func):
+    def inner(*args, **kwargs):
+        with switch_db(BeamlineConfig, 'test_db'), \
```

Comment:
For reasons that I don't recall right now, you can't do that with context managers.  This use case is the only reason `\` line continuation has not been deprecated.